NomaiGrandPrix/SpawnActionFactory.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+
+namespace NomaiGrandPrix
+{
+  public class SpawnActionFactory
+  {
+    private static SpawnActionFactory Instance = new SpawnActionFactory();
+
+
+    private SpawnActionFactory()
+    {
+      // Private constructor
+    }
+
+    public static Action[] GetActionsForSpawn(string spawnId)
+    {
+      if (spawnId == null)
+      {
+        throw new InvalidOperationException("Spawn ID for getting actions cannot be null");
+      }
+      
+      switch (spawnId)
+      {
+        case "Spawn_Module_Sunken":
+          return new Action[] { OpenSunkenModuleAirlock };
+        default:
+          return new Action[0];
+      }
+
+    }
+    private static void OpenSunkenModuleAirlock()
+    {
+      var giantsDeep = Locator._giantsDeep;
+      var airlock = giantsDeep.GetComponentInChildren<NomaiAirlock>();
+      var position = airlock._closeSwitches[0].transform.position;
+      airlock._listInterfaceOrb[0].SetOrbPosition(position);
+    }
+  }
+}
```

Comment:
wyt about 4-space indentation here for consistency

Reply:
fine, even though a bunch of files are already 2-spaced

Reply:
thanks. i was under the impression that it was one other file and the plan was to convert it back to 4-space. i'd advocate for us uniformly sticking to 4, adhering to [the language's layout conventions](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions#layout-conventions).